pkg/kubelet/cm/internal_container_lifecycle.go

```diff
@@ -34,25 +35,53 @@ type InternalContainerLifecycle interface {
 // Implements InternalContainerLifecycle interface.
 type internalContainerLifecycleImpl struct {
 	cpuManager cpumanager.Manager
+	topologyManager topologymanager.Manager
 }
 
 func (i *internalContainerLifecycleImpl) PreStartContainer(pod *v1.Pod, container *v1.Container, containerID string) error {
 	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.CPUManager) {   
-		return i.cpuManager.AddContainer(pod, container, containerID)
+		err := i.cpuManager.AddContainer(pod, container, containerID)
+		if err != nil {
+			return err
+		}
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.TopologyManager) {
+		err := i.topologyManager.AddPod(pod, containerID)
+                if err != nil {
+                        return err
+                }
+	}	
 	return nil
 }
 
 func (i *internalContainerLifecycleImpl) PreStopContainer(containerID string) error {
```

Comment:
Do we need to make the removePod call in both the Pre and Post Stop Container?

Reply:
As far as I can see it doesn't really make a difference, I just included it to be consistent with CPU Manager... PreStop is called in killContainer and PostStop is called in removeContainer (both in kubelet/kuberuntime/kuberuntime_container.go). We can just call it in Post, up to you?

Reply:
Yeah I would go with Post - the only reason I thinking that it may be possible that a container is killed and restarted maybe the same entry to the map can apply whereas with remove container is sounds like there is no chance of a restart.